app/js/lib/speech/synthesis.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+const p = Object.freeze({
+  // Properties
+  synthesis: Symbol('synthesis'),
+  supportsSynthesis: Symbol('supportsSynthesis'),
+  voice: Symbol('voice'),
+  initialised: Symbol('initialised'),
+
+  // Methods
+  populateVoice: Symbol('populateVoice'),
+});
+
+export default class SpeechSynthesis {
+  constructor() {
+    const synthesis = window.speechSynthesis;
+
+    this[p.initialised] = false;
+    this[p.supportsSynthesis] = !!synthesis;
+
+    if (this[p.supportsSynthesis]) {
+      this[p.synthesis] = synthesis;
+      this[p.populateVoice]();
+      this[p.synthesis].onvoiceschanged = this[p.populateVoice].bind(this);
+    } else {
+      this[p.synthesis] = null;
+    }
+
+    this[p.voice] = null;
+
+    Object.seal(this);
+  }
+
+  /**
+   * Speak a text aloud.
+   *
+   * @param {string} text
+   */
+  speak(text = '') {
+    if (!text) {
+      return;
+    }
+
+    const utterance = new SpeechSynthesisUtterance(text);
+
+    if (this[p.voice]) {
+      utterance.voice = this[p.voice];
+    }
+    utterance.lang = 'en';
```

Comment:
yeah I thought of this too (i did it in my initial WIP patch) but I thought it could be good enough with only english for this patch.